lib/lib-storage/src/Upload.ts

```diff
@@ -427,6 +438,31 @@ export class Upload extends EventEmitter {
     });
   }
 
+  private __calculatePartSize(targetPartSizeBytes: number, minPartSize: number): number {
+    const calculatedPartSize = Math.floor(targetPartSizeBytes / this.MAX_PARTS);
+    return Math.max(minPartSize, calculatedPartSize);
+  }
+
+  private __validateUploadPart(dataPart: RawDataPart, partSize: number): void {
+    const actualPartSize = byteLength(dataPart.data) || 0;
```

Comment:
the fallback value of `undefined` should be used, since real data may be zero bytes.